interop/client/client.go

```diff
@@ -214,6 +250,22 @@ func main() {
 	if len(*serviceConfigJSON) > 0 {
 		opts = append(opts, grpc.WithDisableServiceConfig(), grpc.WithDefaultServiceConfig(*serviceConfigJSON))
 	}
+	addMd, err := parseAdditionalMetadataFlag()
+	if err != nil {
+		logger.Fatalf("Error parsing --additional_metadata flag: %v", err)
+	}
+	if addMd != nil {
+		unaryAddMd := func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+			ctx = metadata.NewOutgoingContext(ctx, addMd)
```

Comment:
You should probably be using `metadata.AppendToOutgoingContext()` instead in case a test case ever needs to set its own metadata.  Unfortunately that takes a key/value pair slice instead of the metadata map so will need a slight reworking of `parseAddlMDFlag`.